src/controllers/scripting/legacy/controllerscriptenginelegacy.h

```diff
@@ -114,9 +120,9 @@ class ControllerScriptEngineLegacy : public ControllerScriptEngineBase {
     QHash<QString, std::shared_ptr<ControllerRenderingEngine>> m_renderingScreens;
     // Contains all the scenes loaded for this mapping. Key is the scene
     // identifier (LegacyControllerMapping::ScreenInfo::identifier), value in
-    // the QML root item
-    QHash<QString, std::shared_ptr<QQuickItem>> m_rootItems;
-    QHash<QString, TransformScreenFrameFunction> m_transformScreenFrameFunctions;
+    // the QML root item. Note that the pointer is owned by the QML scene which
+    // will free them on shutdown (ControllerScriptEngineLegacy::shutdown)
+    QHash<QString, mixxx::qml::QmlMixxxControllerScreen*> m_rootItems;
```

Comment:
lets avoid introducing manual memory management. You can wrap this in a unique_ptr. If that doesn't work because QHash needs the key to be copy-able, use `std::unordered_map` instead.

Reply:
Any reason why introducing a smart pointer on a Qt-owned reference here? Just in case you've missed the comment, the pointer is managed by the QML scene, we don't have any control over it.

Reply:
Ah, I just spotted it now. Well the problem is that we do call `delete` on it. So its very easy to mistake it as properly owning. I also fear that the pointer can easily outlive the RenderingEngine their tied to. Ideally the pointers would stay entirely in their rendering engine and in order to access them, you need to access the rendering engine.

Reply:
So sometimes you do delete them and sometimes you don't. It would be really nice if you could fix those ownership issues here.

Reply:
ah, yeah. I was interpreting the 
```cpp
rootItem->setParentItem(pScreen->quickWindow()->contentItem());
```
line as transferring ownership, but thats obviously wrong on second look.